provider/core/src/resource.rs

```diff
@@ -489,6 +568,12 @@ impl fmt::Display for ResourceOptions {
     }
 }
 
+impl AsMut<ResourceOptions> for ResourceOptions {
+    fn as_mut(&mut self) -> &mut ResourceOptions {
+        self
+    }
+}
+
```

Comment:
Curious what this is needed for?

Reply:
There are bounds in adapters/src/fallback/mod.rs that use `AsMut<ResourceOptions>` (rather than `&mut ResourceOptions`), and for whatever reason `AsMut<T>` isn't blanket-impled for all `T`

Reply:
Also see the conversation in https://github.com/unicode-org/icu4x/pull/2115#discussion_r906752080